xc/device.ml

```diff
@@ -1671,7 +1692,10 @@ module Dm_Common = struct
         ]) info.disks in
 
     let restorefile = sprintf qemu_restore_path domid in
-    let disp_options, wait_for_port = cmdline_of_disp info in
+    let disp_options, wait_for_port = if domid_for_vnc
```

Comment:
How does this work? `domid` is an argument that has no default. So how can `cmdline_of_disp info ~domid` and `cmdline_of_disp info` return even the same type?